calico-vpp-agent/cni/storage/storage.go

```diff
@@ -238,8 +244,29 @@ func (hp *HostPortBinding) String() string {
 	return s
 }
 
+/* 8 base64 character hash */
+func hash(text string) string {
+	h := sha512.Sum512([]byte(text))
+	return base64.StdEncoding.EncodeToString(h[:])[:vrfTagHashLen]
+}
+
+func truncateStr(text string, size int) string {
+	if len(text) > size {
+		return text[:size]
+	}
+	return text
+}
+
+func (ps *LocalPodSpec) GetVrfTag(ipFamily vpplink.IpFamily) string {
+	h := hash(fmt.Sprintf("%s%s", ps.NetnsName, ipFamily.ShortStr))
+	s := fmt.Sprintf("%s-%s-%s", h, ipFamily.ShortStr, filepath.Base(ps.NetnsName))
+	return truncateStr(s, MaxApiTagLen)
+}
+
 func (ps *LocalPodSpec) GetInterfaceTag(prefix string) string {
-	return fmt.Sprintf("%s-%s-%s", prefix, ps.NetnsName, ps.InterfaceName)
+	h := hash(fmt.Sprintf("%s-%s-%s", prefix, ps.NetnsName, ps.InterfaceName))
+	s := fmt.Sprintf("%s-%s-%s", h, ps.InterfaceName, filepath.Base(ps.NetnsName))
```

Comment:
Also for the multi-net case, I think it may facilitate troubleshooting if we added the interface name to the tag (before the NetnsName so it doesn't truncated, it would be annoying if it got truncated to just eth ;) )

Reply:
Same